receiver/splunksearchapireceiver/receiver_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunksearchapireceiver
+
+import (
+	"context"
+	"encoding/xml"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+var (
+	logger        = zap.NewNop()
+	config        = &Config{}
+	settings      = component.TelemetrySettings{}
+	id            = component.ID{}
+	ssapireceiver = newSSAPIReceiver(logger, config, settings, id)
+)
+
+func TestPolling(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	cfg.JobPollInterval = 1 * time.Second
+	ssapireceiver.config = cfg
+
+	client := &mockLogsClient{}
+	ssapireceiver.client = client
+
+	file := filepath.Join("testdata", "logs", "testPollJobStatus", "input.xml")
+	client.On("GetJobStatus", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(client.loadTestStatusResponse(t, file), nil)
+	cancelCtx, cancel := context.WithCancel(context.Background())
+	ssapireceiver.cancel = cancel
+	ssapireceiver.checkpointRecord = &EventRecord{}
+
+	err := ssapireceiver.pollSearchCompletion(cancelCtx, "123456")
+	require.NoError(t, err)
+
+	client.AssertCalled(t, "GetJobStatus", "123456")
+
+	err = ssapireceiver.Shutdown(context.Background())
+	require.NoError(t, err)
+}
+
+func TestIsSearchCompleted(t *testing.T) {
+	jobResponse := SearchStatusResponse{
+		Content: Content{
+			Dict: Dict{
+				Keys: []Key{
+					{
+						Name:  "dispatchState",
+						Value: "DONE",
+					},
+				},
+			},
+		},
+	}
+
+	emptyResponse := SearchStatusResponse{}
+
+	done := ssapireceiver.isSearchCompleted(jobResponse)
+	require.True(t, done)
+
+	jobResponse.Content.Dict.Keys[0].Value = "RUNNING"
+	done = ssapireceiver.isSearchCompleted(jobResponse)
+	require.False(t, done)
+
+	done = ssapireceiver.isSearchCompleted(emptyResponse)
+	require.False(t, done)
+}
+
+func TestInitCheckpoint(t *testing.T) {
```

Comment:
Are these supposed to be implemented?

Reply:
Somehow those changes didn't get added in my last commit. Should be good now.